pytorch_lightning/core/optimizer.py

```diff
@@ -54,7 +57,10 @@ def optimizer(self) -> Optimizer:
         return self._optimizer
 
     def _on_trainer_init(self, trainer: "pl.Trainer") -> None:
-        self._trainer = proxy(trainer)
+        if isinstance(trainer, weakref.ProxyType):
```

Comment:
Could you add a comment for this?

Reply:
sure. although I don't really understand why it's needed and it wasn't needed before 🙁